compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1076,20 +1076,24 @@ pub struct VarDebugInfo<'tcx> {
 // BasicBlock
 
 rustc_index::newtype_index! {
-    /// The unit of the MIR [control-flow graph][CFG].
+    /// A node in the MIR [control-flow graph][CFG].
     ///
     /// There are no branches (e.g., `if`s, function calls, etc.) within a basic block, which makes
     /// it easier to do [data-flow analyses] and optimizations. Instead, branches are represented
     /// as an edge in a graph between basic blocks.
     ///
     /// Basic blocks consist of a series of [statements][Statement], ending with a
-    /// [terminator][Terminator]. Basic blocks can have multiple predecessors and successors.
+    /// [terminator][Terminator]. Basic blocks can have multiple predecessors and successors,
+    /// however there is a MIR pass ([`CriticalCallEdges`]) that removes *critical edges*, which
+    /// are edges that go from a multi-successor node to a multi-predecessor node. This pass is
+    /// needed because some analyses require that there are no critical edges in the CFG.
     ///
     /// Read more about basic blocks in the [rustc-dev-guide][guide-mir].
     ///
     /// [CFG]: https://rustc-dev-guide.rust-lang.org/appendix/background.html#cfg
     /// [data-flow analyses]:
     ///     https://rustc-dev-guide.rust-lang.org/appendix/background.html#what-is-a-dataflow-analysis
+    /// [`CriticalCallEdges`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/transform/add_call_guards/enum.AddCallGuards.html#variant.CriticalCallEdges
```

Comment:
Should I use the absolute link or a relative link?

Reply:
I'd use an intra-doc link.
```suggestion
    /// [`CriticalCallEdges`]: rustc_mir::transform::add_call_guards::AddCallguards::CriticalCallEdges
```
(possibly this can be shorter if it's imported)

Reply:
It's a downstream (or upstream? I can't figure out which one 😄) crate though. Aren't those not supported yet?

Reply:
Oh booo. Then yeah an absolute link seems fine, I don't expect anyone to ever look at the stable version of these docs.

>  Aren't those not supported yet?

https://github.com/rust-lang/rust/issues/74481

Reply:
*Is* there even a stable version of the rustc API docs?

Reply:
Apparently yes: https://doc.rust-lang.org/stable/nightly-rustc/rustc_mir/transform/add_call_guards/enum.AddCallGuards.html#variant.CriticalCallEdges

Reply:
Wow, that's bizarre! I guess it makes sense. It's weird though that the URL is /stable/nightly-rustc though :)

Probably not worth it, but maybe it would be better as /stable/rustc-api or something.

Reply:
We don't use absolute links unless necessary, so please make it relative if possible.